File: 2019校赛/domjudge/accounts_gen.cpp
#include <bits/stdc++.h>
using namespace std;
const int len = 6;

char orand(){
	if(rand() % 36 < 26){
		return rand() % 26 + 'a' - (rand()&1) * 32;
	}
	return rand() % 10 + '0';
}

int main(){
	srand(time(NULL));
	printf("accounts\t1\n");
	for(int id = 1; id <= 80; ++ id){
		// printf("team\tt%d\tteam%03d\t", id+6, id);
		printf("team\tteam%03d\tteam%03d\t", id, id);
		// printf("team%03d", id);
		for(int i = 0; i < len; ++ i){
			printf("%c", orand());
		}
		printf("\n");
	}
	return 0;
}

File: 2019校赛/domjudge/accounts_gen.py
#!/usr/bin/python

import random

def randchar():
	return random.choice("1234567890QWERTYUIOPASDFGHJKLZXCVBNMqwertyuiopasdfghjklzxcvbnm");
def randpasswd(length = 6):
	ret = ""
	for i in range(length):
		ret += randchar();
	return ret;

teamnames = [];
fname = "team.names";
with open(fname) as f:
	tmp = f.read().splitlines();
	for name in tmp:
		name = name.strip();
		if(len(name) == 0):
			continue;
		teamnames.append(name);

# for name in teamnames:
# 	print(name);

fname = "accounts.tsv";
with open(fname, 'w') as f:
	f.write("accounts\t1");

with open(fname, 'a') as f:
	for i in range(len(teamnames)):
		f.write("\nteam\t%s\tteam%02d\t%s" % (teamnames[i], i+1, randpasswd()));
